Summary

Wikitelmex opened the pull request:
- add person generic class
- add student class inherits from person
- add teacher class inherits from person
- fix linters

wikitelmex1 committed:
Summary

- add person generic class
- add student class inherits from person
- add teacher class inherits from person
- fix linters

wikitelmex1 committed:
Update README.md

- fit Readme.md to microverse requirements

D4nielJ commented on person.rb:
```diff
@@ -0,0 +1,34 @@
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    @parent_permission
+  end
```
- [x] I think there is a mistake in this method. These are the requirements:

> Public method can_use_services? that returns true if person is of age or if they have permission from parents.

Kindly please fix this.

D4nielJ reviewed (changes requested):
## Hello @Wikitelmex, 

You're doing great! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

<img src="https://media4.giphy.com/media/KyBesf8kNT6s88SEgb/giphy.gif"/>

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

## Happy coding and stay awesome! 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

wikitelmex1 committed:
changes required 1

- fix can_use_services? returns true if older than 18 or have paren permission, before only true if it had permission

D4nielJ commented on person.rb:
```diff
@@ -23,7 +23,7 @@ def initialize(age, name: 'Unknown', parent_permission: true)
   end
 
   def can_use_services?
-    @parent_permission
+    @parent_permission || @age >= 18
   end
```
[OPTIONAL] Good job implementing the last change I required. For maximum results, it could be better if you use the `of_age?` method that we have already defined. This way if we ever need to change the way the method is calculated we only have to do it in one place.

This follows principles as DRY, which you are already familiar with, or Single Responsibility Principle, which you will be studying more in-depth in the future.

Wikitelmex replied:
- thanks a lot, and good observation, was my mistake, however, it was solved and calls that function.
````
  def can_use_services?
    @parent_permission || of_age?
  end
````

D4nielJ reviewed (approved):
## Hello @Wikitelmex, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

![](https://i.giphy.com/media/U4DswrBiaz0p67ZweH/giphy.webp)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

wikitelmex1 committed:
Update person.rb

- calls private method of_age that already had calculation of over legal age

wikitelmex1 committed:
Update person.rb
